test/src/main/java/io/strimzi/test/BaseITST.java

```diff
@@ -0,0 +1,140 @@
+
+    protected String testClass;
+    protected String testName;
+
+    private void applyClusterOperatorInstallFiles() {
+        TimeMeasuringSystem.setTestName(testClass, testClass);
+        TimeMeasuringSystem.startOperation(Operation.CO_CREATION);
+        clusterOperatorMap = Arrays.stream(new File(CO_INSTALL_DIR).listFiles()).sorted().filter(file ->
+                !file.getName().matches(".*(Binding|Deployment)-.*")
+        ).collect(Collectors.toMap(file -> file, f -> TestUtils.getContent(f, TestUtils::toYamlString), (x, y) -> x, LinkedHashMap::new));
+        for (Map.Entry<File, String> entry : clusterOperatorMap.entrySet()) {
+            LOGGER.info("Applying configuration file: {}", entry.getKey());
+            KUBE_CLIENT.clientWithAdmin().applyContent(entry.getValue());
+        }
+        TimeMeasuringSystem.stopOperation(Operation.CO_CREATION);
+    }
+
+    private void deleteClusterOperatorInstallFiles() {
+        TimeMeasuringSystem.setTestName(testClass, testClass);
+        TimeMeasuringSystem.startOperation(Operation.CO_DELETION);
+        for (Map.Entry<File, String> entry : clusterOperatorMap.entrySet()) {
+            LOGGER.info("Removing configuration file: {}", entry.getKey());
+            KUBE_CLIENT.clientWithAdmin().deleteContent(entry.getValue());
+        }
+        TimeMeasuringSystem.stopOperation(Operation.CO_DELETION);
+    }
+
+    private void createNamespaces(String useNamespace, List<String> namespaces) {
+        deploymentNamespaces = namespaces;
+        for (String namespace: namespaces) {
+            LOGGER.info("Creating namespace: {}", namespace);
+            KUBE_CLIENT.createNamespace(namespace);
+            KUBE_CLIENT.waitForResourceCreation("Namespace", namespace);
+            LOGGER.info("Namespace {} created", namespace);
+        }
+        LOGGER.info("Using namespace {}", useNamespace);
+        KUBE_CLIENT.namespace(useNamespace);
+    }
+
+    protected void createNamespaces(String useNamespace) {
+        createNamespaces(useNamespace, Collections.singletonList(useNamespace));
+    }
+
+    protected void deleteNamespaces() {
+        for (String namespace: deploymentNamespaces) {
+            LOGGER.info("Deleting namespace: {}", namespace);
+            KUBE_CLIENT.deleteNamespace(namespace);
+            KUBE_CLIENT.waitForResourceDeletion("Namespace", namespace);
+            LOGGER.info("Namespace {} deleted", namespace);
+        }
+        LOGGER.info("Using namespace {}", CLUSTER.defaultNamespace());
+        KUBE_CLIENT.namespace(CLUSTER.defaultNamespace());
+    }
+
+    protected void createCustomResources(List<String> resources) {
+        deploymentResources = resources;
+        for (String resource : resources) {
+            LOGGER.info("Creating resources {}", resource);
+            KUBE_CLIENT.clientWithAdmin().create(resource);
+        }
+    }
+
+    protected void deleteCustomResources() {
+        for (String resource : deploymentResources) {
```

Comment:
To make it more resilient it's better to process the list in reverse order. That way you avoid problems dependencies between resources where you can't delete resource A because resource B still depends on it. TBH I'm not sure whether we have anything like that, but I would still process in reverse order.

Reply:
I agree, I added list reservation before deletion. Same for namespaces.